bin/varnishd/cache/cache_deliver_proc.c

```diff
@@ -136,9 +133,18 @@ VDP_bytes(struct vdp_ctx *vdc, enum vdp_action act,
 	return (vdc->retval);
 }
 
+/*
+ * oc: optional, only passed to the first VDP
+ * req: optional, for client-side only VDPs
+ * hd: headers to modify (request or response)
+ * cl: known content-length or -1, settable
+ */
+
 int
-VDP_Push(VRT_CTX, struct vdp_ctx *vdc, struct ws *ws, const struct vdp *vdp,
-    void *priv)
+VDP_Push(VRT_CTX, struct vdp_ctx *vdc, struct ws *ws,
+     const struct vdp *vdp, void *priv,
+     struct objcore *oc, struct req *req,
+     struct http *hd, intmax_t *cl)
```

Comment:
Nit pick, the `priv` argument tends to be first in callbacks and last otherwise (though not always).

Reply:
I did ponder this question for a bit: The `priv` argument is the `priv` member from `struct vdp_entry`, so it belongs to the `vdp`. Thus I kept it next to it.